core/src/main/java/org/elasticsearch/search/SearchModule.java

```diff
@@ -292,16 +287,23 @@ public void registerSuggester(String key, Suggester<?> suggester) {
     }
 
     /**
-     * Register a new ScoreFunctionParser.
+     * Register a new ScoreFunctionBuilder. Registration does two things:
+     * <ul>
+     * <li>Register the {@link ScoreFunctionParser} which parses XContent into a {@link ScoreFunctionBuilder} using its {@link ParseField}
+     * </li>
+     * <li>Register the {@link Writeable.Reader} which reads a stream representation of the builder under a single name. This name must be
+     * the same name as the builder returns for getWriteableName.</li>
+     * </ul>
      */
-    public <T extends ScoreFunctionBuilder<T>> void registerScoreFunction(ScoreFunctionParser<T> parser, Writeable.Reader<T> reader) {
-        for (String name: parser.getNames()) {
+    public <T extends ScoreFunctionBuilder<T>> void registerScoreFunction(ParseField xContentField, ScoreFunctionParser<T> parser,
```

Comment:
can you call the first paramenter "functionName" or something along those lines?